effect/README.md

```diff
@@ -0,0 +1,294 @@
+ * **Concurrency**. `IO` has all the concurrency features of `Future`, and more, based on a clean fiber concurrency model designed to scale well past the limits of native threads. Unlike other effect monads, `IO`'s concurrency primitives do not leak threads.
+ * **Interruptibility**. All concurrent computations can be interrupted, in a way that still guarantees resources are cleaned up safely, allowing you to write aggressively parallel code that doesn't waste valuable resources or bring down production servers.
+ * **Resource Safety**. `IO` provides composable resource-safe primitives that ensure resources like threads, sockets, and file handles are not leaked, which allows you to build long-running, robust applications. These applications will not leak resources, even in the presence of errors or interruption.
+ * **Immutability**. `IO`, like Scala's immutable collection types, is an immutable data structure. All `IO` methods and functions return new `IO` values. This lets you reason about `IO` values the same way you reason about immutable collections.
+ * **Reification**. `IO` reifies programs. In non-functional Scala programming, you cannot pass programs around or store them in data structures, because programs are not values. But `IO` turns your programs into ordinary values, and lets you pass them around and compose them with ease.
+ * **Performance**. Although simple, synchronous `IO` programs tend to be slower than the equivalent imperative Scala, `IO` is extremely fast given all the expressive features and strong guarantees it provides. Ordinary imperative Scala could not match this level of expressivity and performance without tedious, error-prone boilerplate that no one would write in real-life.
+
+While functional programmers *must* use `IO` (or something like it) to represent effects, nearly all programmers will find the features of `IO` help them build scalable, performant, concurrent, and leak-free applications faster and with stronger correctness guarantees than legacy techniques allow.
+
+Use `IO` because it's simply not practical to write real-world, correct software without it.
+
+# Usage
+
+## Main
+
+Your main function can extend `SafeApp`, which provides a complete runtime
+system and allows your entire program to be purely functional.
+
+```scala
+import scalaz.effect.{IO, SafeApp}
+import scalaz.effect.console._
+
+object MyApp extends SafeApp {
+  def run(args: IList[String]): IO[Unit] =
+    for {
+      _ <- putStrLn("Hello! What is your name?")
+      n <- getStrLn
+      _ <- putStrLn("Hello, " + n + ", good to meet you!")
+    } yield ()
+}
+```
+
+## Pure Values
+
+You can lift pure values into `IO` with `IO.apply`:
+
+```scala
+val liftedString: IO[String] = IO("Hello World")
+```
+
+The constructor uses non-strict evaluation, so the parameter will not be evaluated until when and if the `IO` action is executed at runtime.
```

Comment:
Poor wording on my part. It never results in duplicated computation. If you use it for effects, the results are undefined, and could include runtime errors (e.g. `IO(throw e).map(_ => 1)`).

The name of this could be `IO.point`. But then I'd want to delete `IO.apply` because it is abused for effectful code but that abuse is not supported by `IO`.

`point` may never be used for effects, the only question is whether or not `IO.apply` should be used for `point` (as in 7), or whether it should not be used at all to force users to be explicit.

Reply:
I think it's best to just remove `IO.apply` and have it named maybe `IO.point` or `IO.lazyPure` or something.